### PR TITLE
feat: add wellknown api

### PR DESCRIPTION
**Pull request by skynet2**
closes: https://github.com/trustbloc/vcs/issues/839

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: pkg/restapi/v1/devapi/controller.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	if scheme := ctx.Request().URL.Scheme; len(scheme) > 0 {
+		contextURL.WriteString(scheme + "://")
+	} else {
+		contextURL.WriteString("https://")
+	}
+
+	if hostName := ctx.Request().URL.Hostname(); len(hostName) > 0 {
+		contextURL.WriteString(hostName + "/")
+	} else {
+		contextURL.WriteString("localhost")
```
Comment (aholovko) [inline pkg/restapi/v1/devapi/controller.go:61]: this looks strange in production code, plz elaborate on this logic

File: pkg/service/didconfiguration/didconfiguration_service.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+		if err != nil {
+			return nil, resterr.NewValidationError(resterr.SystemError, "profileID",
+				err)
+		}
+		cred.Issuer = verifiable.Issuer{
+			ID: profile.SigningDID.DID,
+		}
+		cred.Subject = map[string]interface{}{
+			"id":     profile.SigningDID.DID, // todo nothing in JSON ???
+			"origin": fmt.Sprintf("%s://%s", u.Scheme, u.Hostname()),
```
Comment (aholovko) [inline pkg/service/didconfiguration/didconfiguration_service.go:117]: shouldn't this be an URL from profile?

File: test/bdd/pkg/vc-devapi/vc_devapi_steps.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+func (s *Steps) httpGet(profileType string, id string, credType string) error {
+	s.credType = credType
+
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet,
+		fmt.Sprintf("http://localhost:8075/%s/profiles/%s/well-known/did-config",
+			profileType, id), http.NoBody)
+
+	req.Header.Add("X-API-Key", "rw_token")
+
+	if err != nil {
```
Comment (aholovko) [inline test/bdd/pkg/vc-devapi/vc_devapi_steps.go:60]: move error checking before `req.Header.Add`

File: test/bdd/pkg/vc-devapi/vc_devapi_steps.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	return &Steps{bddContext: ctx}
+}
+
+// RegisterSteps registers VC scenario steps.
+func (s *Steps) RegisterSteps(sc *godog.ScenarioContext) {
+	sc.Step(`^I request did config for "([^"]*)" with ID "([^"]*)" and type "([^"]*)"$`, s.httpGet)
+	sc.Step(`^I receive response with status code "([^"]*)" for didconfig$`, s.checkResponseStatus)
+}
+
+func (s *Steps) checkResponseStatus(status string) error {
```
Comment (aholovko) [inline test/bdd/pkg/vc-devapi/vc_devapi_steps.go:36]: can we perform some basic checks for didconfig from response as well? like some properties that for sure should be there...

**Review by aholovko — CHANGES_REQUESTED** [review]

File: pkg/restapi/v1/devapi/controller.go (hunk 1 above)
Comment (fqutishat) [reply pkg/restapi/v1/devapi/controller.go:61]: I agree with that

Comment (skynet2) [reply pkg/restapi/v1/devapi/controller.go:61]: We need to build absolute URI. in HTTP request, standard Request().URL there is no possibility of receiving full URL (RequestURI() in some cases can return only Path, without schema and domain, so I combine URL by myself from request, that code basically can be moved to some helper. cc @aholovko @fqutishat

Comment (aholovko) [reply pkg/restapi/v1/devapi/controller.go:61]: Stas, you are using that URL as a value of @context - I think it's not correct, there should be constant value "https://identity.foundation/.well-known/did-configuration/v1" (JSON-LD context for "DID Configuration" document). Probably, you confused that with https://identity.foundation/.well-known/resources/did-configuration/#did-configuration-uri that describes where that document should be hosted. cc @skynet2

File: test/bdd/pkg/vc-devapi/vc_devapi_steps.go (hunk 4 above)
Comment (skynet2) [reply test/bdd/pkg/vc-devapi/vc_devapi_steps.go:36]: fixed

File: pkg/restapi/v1/devapi/controller.go (hunk 1 above)
Comment (skynet2) [reply pkg/restapi/v1/devapi/controller.go]: you are right @aholovko , fixed

**Pull request merged**